src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionTelemetry.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Extensions.Internal;
+
+namespace System.Net
+{
+    [EventSource(Name = "System.Net.NameResolution")]
+    internal sealed class NameResolutionTelemetry : EventSource
+    {
+        public static readonly NameResolutionTelemetry Log = new NameResolutionTelemetry();
+
+        public static new bool IsEnabled => Log.IsEnabled();
+
+        private const int ResolutionStartEventId = 1;
+        private const int ResolutionSuccessEventId = 2;
+        private const int ResolutionFailureEventId = 3;
+
+        private PollingCounter? _lookupsRequestedCounter;
+        private EventCounter? _lookupsDuration;
+
+        private long _lookupsRequested;
+
+        protected override void OnEventCommand(EventCommandEventArgs command)
+        {
+            if (command.Command == EventCommand.Enable)
+            {
+                // The cumulative number of name resolution requests started since the process started.
+                _lookupsRequestedCounter ??= new PollingCounter("dns-lookups-requested", this, () => Interlocked.Read(ref _lookupsRequested))
+                {
+                    DisplayName = "DNS Lookups Requested"
+                };
+
+                _lookupsDuration ??= new EventCounter("dns-lookups-duration", this)
+                {
+                    DisplayName = "Average DNS Lookup Duration",
+                    DisplayUnits = "ms"
+                };
+            }
+        }
+
+        private const int MaxIPFormattedLength = 128;
+
+        [Event(ResolutionStartEventId, Level = EventLevel.Informational)]
+        public ValueStopwatch ResolutionStart(string hostNameOrAddress)
```

Comment:
You should only add an event with a 'Start' suffix if you pair it with an event with a 'Stop' suffix. ActivityId threading will be messed up otherwise. If you don't need ActivityID threading at this level, you should use a different name for the event.

Reply:
I renamed the event from `ResolutionStart` to `ResolutionStarting`.